libs/@local/hash-isomorphic-utils/src/util.ts

```diff
@@ -14,6 +14,13 @@ export const isUnknownObject = (
 ): x is { [key in PropertyKey]: unknown } =>
   x !== null && typeof x === "object";
 
+/**
+ * Ensures that the array has at least one element. If not it returns `undefined`.
+ */
+export const atLeastOne = <T>(array: T[]): [T, ...T[]] | undefined =>
+  // @ts-expect-error –– @see https://github.com/microsoft/TypeScript/issues/29841
+  array.length > 0 ? (array satisfies [T, ...T[]]) : undefined;
+
```

Comment:
Is this supposed to be here? It also already exists in `type-system`

Reply:
Whoops, that's a rebase error. Fixed in [`f384e79` (#4688)](https://github.com/hashintel/hash/pull/4688/commits/f384e79d94c6a317a81672eb4f443239b584cc5c)